client/state/navigation/middleware.js

```diff
@@ -47,8 +47,13 @@ export const navigationMiddleware = store => {
 				return next( action );
 			case HISTORY_REPLACE:
 				if ( action.path ) {
-					// Replace the history entry, but don't dispatch a new navigation context.
-					page.replace( action.path, null, false, false );
+					if ( action.saveContext ) {
+						// Replace the history entry, but don't dispatch a new navigation context.
+						//pate, state, init, dispatch
```

Comment:
is `pate` a typo? `path` maybe?

as above, I think inline comments can make this even more clear…

```js
page.replace(
	action.path, // path
	null, // state to insert into history context
	false, // whether or not to initialize context??
	false, // don't dispatch the update - change the URL but don't navigate
);
``

Reply:
Good catch, I'll update.